docs/export_results.rst

```diff
@@ -0,0 +1,180 @@
+- Though Taichi now supports exporting to .png, .jpg and .bmp these 3 mainstream formats, we strongly **recommend using .png** which has been tested well at this moment. For example:
+
+.. code-block:: python
+    
+    import taichi as ti
+    import os
+    pixel = ti.var(ti.u8, shape=(512, 512, 3))
+
+    @ti.kernel
+    def paint():
+        for I in ti.grouped(pixel):
+            pixel[I] = ti.random() * 255
+
+    iterations = 1000
+    gui = ti.GUI("Export Result", res = 512)
+
+    # set up the images output
+    output_dir = "./results/"
+    if os.path.exists(output_dir) == False:
+        os.makedirs(output_dir)
+
+    # mainloop
+    for i in range(iterations):
+        paint()
+        gui.set_image(pixel)
+
+        # img_name = os.path.join(output_dir, f"frame_{i}.jpg")   # create filename with suffix jpg
+        # img_name = os.path.join(output_dir, f"frame_{i}.bmp")   # create filename with suffix bmp
+        img_name = os.path.join(output_dir, f"frame_{i}.png")   # create filename with suffix png
+        print("%s recorded in frame %d" % (img_name, i))
+        gui.show(img_name)  # export and show in GUI
+
+- After running these codes above, you should get a batch of images in your "./results" folder.
+
+Export Images by Taichi built-ins
++++++++++++++++++++++++++++++++++
+
+- Why we have to instantiate GUI just for the sake of export images? The former method under the help of "ti.GUI" looks weird.
+- If you are familiared with Taichi, we recommend using these following methods for manipulating images.
+- This second approach is to use **ti.imwrite**. It is much more flexible compared with its former.
+
+.. function:: ti.imwrite(img, filename)
+
+    :parameter img: (Matrix or Expr) the images you want to export
+    :parameter filename: (string) filename you want to save
+
+    Please make sure that the input image is a Taichi Matrix or Expr and **it has shape [height, width, compoents] correctly**. For example:
+    
+    .. code-block:: python
+
+        import taichi as ti
+        pixel = ti.var(ti.u8, shape=(512, 512, 3))
+
+        @ti.kernel
+        def set_pixel():
+            for I in ti.grouped(pixel):
+                pixel[I] = ti.random() * 255
+        
+        set_pixel()
+        ti.imwrite(pixel.to_numpy(), f"imwrite_export.png")
+
+.. function:: ti.imread(filename, channels = 0)
+
+    :parameter filename: (string) filename of the image you want to load
+    :parameter channels: (int) the number of channels in your specified image, the default value is zero.
+    
+    This function can load an image from the target filename, return it as a np.ndarray
+
+.. function:: ti.imshow(img, windname)
+
+    :parameter img: (Matrix or Expr) the image you want to show in the GUI
+    :parameter windname: (string) the name of GUI window
+
+    This function will open an instance of ti.GUI and show the input image on the screen.
```

Comment:
These are not about `export_results`, how about to move these API-reference-styled stuffs to gui.rst and left a link to export_results there? Let's keep focus on `export` instead of `import` here.

Reply:
Totally Agree.To be honest, I think this part divagate as well.So I plan to expand these descriptions then move all of them to the end of gui.rst.How about it?

Reply:
Sounds not so good to me, gui.rst is mainly for the `ti.GUI`, not for `ti.imshow` and `ffmpeg`.

Reply:
Ummm, Maybe I havn't get your point clearly now. How would it be better to do with these `ti.imread()` and `ti.imshow()`? Simply Delete them OR moving them to somewhere else? :O

Reply:
Let's move `ti.imread`, `ti.imwrite`, `ti.imshow` to gui.rst, and then add a link to `ti.imwrite` in export_results.rst, wdyt?